### PR TITLE
Unlooping Golden File Tests

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This change unloops golden file tests in the framework. Looping golden file tests like this causes issues when the first expectation fails, it will not execute the rest, meaning they have to be run multiple times to iterate through the expectations.

## Tests

Un-tangled

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/material/text_field_test.dart [hunk 1]
```diff
@@ -681,7 +681,7 @@ void main() {
 
     await expectLater(
       find.byType(TextField),
-      matchesGoldenFile('text_field_cursor_width_test.0.0.png', version: 0),
+      matchesGoldenFile('text_field_cursor_width_test.0.png'),
     );
     EditableText.debugDeterministicCursor = false;
   }, skip: !isLinux);
```
Comment (Piinks) [inline packages/flutter/test/material/text_field_test.dart]:
```suggestion
  });
```

File: packages/flutter/test/material/text_field_test.dart [hunk 2]
```diff
@@ -703,10 +703,10 @@ void main() {
 
     await expectLater(
       find.byType(TextField),
-      matchesGoldenFile('text_field_cursor_width_test.1.0.png', version: 0),
+      matchesGoldenFile('text_field_cursor_width_test.1.png'),
     );
     EditableText.debugDeterministicCursor = false;
-  }, skip: !isLinux);
```
Comment (Piinks) [inline packages/flutter/test/material/text_field_test.dart]: Looks like this was reintroduced when a change was accidentally merged to master.

File: packages/flutter/test/widgets/shadow_test.dart [hunk 3]
```diff
@@ -99,34 +101,39 @@ void main() {
...
-              color: Colors.yellow[200],
-              child: PhysicalShape(
-                color: Colors.green[900],
-                clipper: ShapeBorderClipper(shape: BeveledRectangleBorder(borderRadius: BorderRadius.circular(20.0))),
-                elevation: elevation,
-                child: const SizedBox(
-                  height: 100.0,
-                  width: 100.0,
-                ),
+
```
Comment (Piinks) [inline packages/flutter/test/widgets/shadow_test.dart]:
```suggestion
```

**Comment by skia-gold** [comment]
Gold has detected one or more untriaged digests on patchset 3.
Please triage them at https://flutter-gold.skia.org/search?issue=51883.

File: packages/flutter/test/material/bottom_navigation_bar_test.dart [hunk 4]
```diff
@@ -1404,54 +1404,53 @@ void main() {
...
+      testWidgets('pump $pump', (WidgetTester tester) async {
+        await tester.pumpWidget(widget);
+        await tester.tap(find.text('Green'));
+
+        await tester.pump(Duration(milliseconds: 30 * pump));
+        await expectLater(
+          find.byType(BottomNavigationBar),
+          matchesGoldenFile('bottom_navigation_bar.shifting_transition.${pump - 1}.png'),
+        );
+      }, skip: isBrowser);
```
Comment (Piinks) [inline packages/flutter/test/material/bottom_navigation_bar_test.dart]:
```suggestion
      }, skip: isBrowser); // TODO(yjbanov): web does not support golden tests yet: https://github.com/flutter/flutter/issues/40297
```

File: packages/flutter/test/material/text_field_test.dart (hunk 2 above)
Comment (Piinks) [reply packages/flutter/test/material/text_field_test.dart]: #49572

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by dnfield — APPROVED** [review]

**Pull request merged**